src/lib/RecordChunkIterator.ts

```diff
@@ -59,15 +59,17 @@ export class RecordChunkIterator extends TypedEventManager<IIteratorEvents> {
    * @param err
    */
   public async afterEach(prevChunk: RecordsChunk, err?: PartialRejection | Error): Promise<void> {
-    const rejectedIds = (err instanceof PartialRejection && err?.recordIds) || []
-    if (rejectedIds.length) {
-      await this.api.updateRecordStatus(this.session, rejectedIds, ERecordStatus.REVIEW)
-    }
+    const promises = []
+
+    const rejectedIds = err instanceof PartialRejection ? err.recordIds : []
+    promises.push(this.api.updateRecordStatus(this.session, this.rejectedIds, ERecordStatus.REVIEW))
 
     const acceptedIds = prevChunk.recordIds.filter((recordId) => !rejectedIds.includes(recordId))
-    if (acceptedIds) {
-      await this.api.updateRecordStatus(this.session, acceptedIds, ERecordStatus.ACCEPTED)
-    }
+    promises.push(this.api.updateRecordStatus(this.session, acceptedIds, ERecordStatus.ACCEPTED))
+
+    await Promise.all(promises)
+    this.acceptedIds = this.acceptedIds.concat(acceptedIds)
```

Comment:
is there a place where we need to access all of the `acceptedIds` / `rejectedIds` at the end? curious the reason for still needing to concat here

Reply:
xD only testing purposes